Add CLEAN_ONLY flag

evitalis opened the pull request:
When in an environment where there may be connectivity issues /tmp will be potentially filled by broken downloads. This happens as mktemp create new randomly named directories each time the script is run.

I know @qbit has issue #48 to remove some options but I felt this useful to add. I am not familiar with man page syntax so I did not add it there.

Thoughts? Perhaps it may be better to add cleaning of snap downloads in $DST or /tmp as the first step each time the script is run instead.

qbit commented:
Let me think on this one. I haven't been happy with the mktemp change I made.. specifically because of the random dirs it makes.  It also breaks adding `-C` to `ftp`'s options for re-starting. I currently solved this by setting DST in my conf.. 

I also tend to run `snap`, then copy the verified dist files around to other machines on my network.. so I don't really want to auto-remove after the run..

evitalis commented:
Currently the way I implemented it the cleaning happens on demand, the auto clean was just an idea. What you said makes sense though as to why it should not happen automatically. I did not see a `-C` flag which is why I added it but the letter can be changed to anything really.

qbit commented:
Yeah, the `-C` is specific to `ftp`.

So like I have this in my snap.conf:
```
FTP_OPTS:-C -V
```

Makes it so ftp resumes where it left off when a download fails.

qbit commented on snap:
```diff
@@ -322,19 +323,22 @@ REBOOT=$(get_conf_var 'REBOOT' || echo 'false')
 	echo "cdn.openbsd.org")
 
-while getopts "b:Bc:dD:ehiIkKm:M:nrRsSuUV:Wx" arg; do
+while getopts "Cb:Bc:dD:ehiIkKm:M:nrRsSuUV:Wx" arg; do
```
Can you change this to be on alphabetical order?

evitalis replied:
Went ahead and reordered this in the next commit. Sorry for the delay on it.

qbit commented on snap:
```diff
@@ -408,6 +412,14 @@ while getopts "b:Bc:dD:ehiIkKm:M:nrRsSuUV:Wx" arg; do
+		rm -rf /tmp/snap.*
+	exit 0
+fi
+
```
This should be run somewhere near the end of the script. I would put it just before the reboot logic.

qbit replied:
oh wait. nm, then that doesn't fix the issue you were having :P

evitalis commented:
Merged, thanks for the review.

qbit commented on snap:
```diff
@@ -408,6 +412,14 @@ while getopts "b:Bc:dD:ehiIkKm:M:nrRsSuUV:Wx" arg; do
+		rm -rf $DST/snap.*
+	else
+		rm -rf /tmp/snap.*
+	exit 0
```
I missed it before, but there was a missing `fi` here. Please make sure you run the script after making changes, just to be sure it works!

evitalis commented:
Not sure how I missed that either. Will fix ASAP and get another PR in.

qbit commented:
> Not sure how I missed that either. Will fix ASAP and get another PR in.

No worries - I pushed a fix already (with slightly modified logic and comments explaining it :D)

evitalis commented:
Thanks. Will be extra diligent about testing going forward. Thanks for the explanation and updated logic.
